### ISSUE TITLE
**100% WER on Persian ASR - Using Pre-trained Quartznet15x5**

### ISSUE THREAD

**Issue by soheiltehranipour** [NVIDIA/NeMo#1107]
**How can I get better results in WER for the test set? It's Stuck on 100%**

Hello, 
This issue is for "**Making an ASR model using NeMo for the Persian language**."
I have used the QuartzNet15x5 pretrained checkpoint to **fine-tune** the ASR model on my custom data in Persian lanuage.
I followed the article in NVIDIA's blog ([Link](https://developer.nvidia.com/blog/jump-start-training-for-speech-recognition-models-with-nemo/)) written by @okuchaiev which is used for Russian & Spanish.
Finally, After 500 epochs, I got _4% WER on training data_ but for the evaluation, there is a big problem. (_100% WER on Test data_)
Any help would be appreciated.

Step by step procedure done by me:
1. Import all packages
2. Download dataset from common voice (and then unzip it)
3. change dataset format from mp3 to wav based on the script on the blog.
4. provide train and test manifest (.json) from (.tsv) files.
5. provide new ".yaml" file while changing only the characters in "LABELS" Parameters.
6. `!python quartznet.py --model_config="quartznet15x5-fa.yaml" --train_dataset=.../train.json" --eval_datasets=".../test.json" --batch_size=32 --num_epochs=100--checkpoint_dir='/checkpoint'
`
(I go with 100 for five times using checkpoints)

**Environment overview (please complete the following information)**

 - Environment location: **Google Colab** (with **Tesla T4** GPU)
 - Method of NeMo install: **!pip install nemo_toolkit[asr]**
- NeMo version: **0.11.0**
- Learning Rate: **1e-2**
- Optimizer : **Novograd**

**Environment details**

- OS version : "**Ubuntu18.04.3 LTS**"
- PyTorch version : "**1.5.0+cu101**"
- Dataset used for Fine-Tuning: "**Common Voice**" ([Mozilla](https://commonvoice.mozilla.org/en/datasets)) 

Soheil Tehranipour

**Comment by okuchaiev** [comment]
Hi @soheiltehranipour  - your learning rate is certainly too high. Try 0.001 (or even smaller). 
What we've observed is that during fine-tuning the LR should be ~10x smaller compared to what was used during training from scratch. Also, what is your batch size?
Make sure you've turned off transcript normalization (since you are using non-Eng language).

I am surprised that training WER is actually OK (4%) while evals is bad. Might be that you are actually training from scratch and overfitting. Not fine-tuning.

Can you please switch to the latest NeMo (from Main branch)? ASRModel has ["change_vocabulary" method now](https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/asr/01_ASR_with_NeMo.ipynb#scrollTo=2f142kIQc1Z2). So roughly you would do:
```
# initializes encoder + decoder with pre-trained English model
quartznet = nemo_asr.models.EncDecCTCModel.from_pretrained(model_name="QuartzNet15x5Base-En")
# Change vocabulary to Persian here
quartznet.change_vocabulary(
    new_vocabulary=[...]
)
```
and use PTL for training.
Let us know how that would go.

**Comment by usamaehsan** [comment]
@soheiltehranipour  I am also using colab(p100 gpu) to make asr of my language
Can you please, tell me
How much is your data size?
And how much time it takes in training on your dataset?

**Comment by samabdullah** [comment]
@soheiltehranipour @okuchaiev  I am also trying to set up NeMo ASR for a non-English language using the same tutorial you referred to. I follow the same steps but am not able to locate quartznet.py - can you please help? Is this a file you made or should this be available in the library? 

Sorry if this is a very basic question, I have spent weeks trying to find this file. I have seen it referenced in many issues but unable to locate.

**Comment by soheiltehranipour** [comment]
> 
> 
> @soheiltehranipour @okuchaiev I am also trying to set up NeMo ASR for a non-English language using the same tutorial you referred to. I follow the same steps but am not able to locate quartznet.py - can you please help? Is this a file you made or should this be available in the library?
> 
> Sorry if this is a very basic question, I have spent weeks trying to find this file. I have seen it referenced in many issues but unable to locate.

Dear Samabdullah,
First I should recommend to use the latest version of NeMo as @okuchaiev wrote a lot in this repo's issues which is Version 1.0 with a better functionality. It uses PyTorch Lightning which will help your whole training better and easier.

Second, the requested file can be downloaded here :
[Quartznet.py](https://nemo-public.s3.us-east-2.amazonaws.com/transfer-learning-04-2020/quartznet.py)

Third, You can get more from the link below: 
https://developer.nvidia.com/blog/jump-start-training-for-speech-recognition-models-with-nemo/

**Comment by soheiltehranipour** [comment]
> 
> 
> @soheiltehranipour " how to provide new ".yaml" file while changing only the characters in "LABELS" Parameters. wont it changed the pretrained config file too

Are you using the latest version of NeMo?
You can changed it from the label part.

**Comment by Ashbajawed** [comment]
@soheiltehranipour  how you transported the 3 pt files i.e ( JasperDecoderForCTC-STEP-82000.pt  JasperEncoder-STEP-82000.pt  trainer-STEP-82000.pt) into .nemo format
